app/models/ont/well_factory.rb

```diff
@@ -37,28 +37,17 @@ def build_well(attributes)
       return unless attributes.key?(:samples)
 
       begin
-        @tag_service = create_tag_service(attributes[:samples].count)
+        validate_num_samples(attributes[:samples].count)
         @request_factories = attributes[:samples].map do |request_attributes|
-          RequestFactory.new(well: well,
-                             request_attributes: request_attributes,
-                             tag_service: @tag_service)
+          RequestFactory.new(well: well, request_attributes: request_attributes)
         end
       rescue StandardError => e
         @raised_exceptions << e
       end
     end
 
-    def create_tag_service(num_samples)
-      case num_samples
-      when 1
-        ::TagService.new(nil)
-      when 96
-        ::TagService.new(::TagSet.find_by!(name: 'OntWell96Samples'))
-      when 384
-        ::TagService.new(::TagSet.find_by!(name: 'OntWell384Samples'))
-      else
-        raise "'#{num_samples}' is not a supported number of samples"
-      end
+    def validate_num_samples(num_samples)
+      raise "'#{num_samples}' is not a supported number of samples" unless num_samples == 1
```

Comment:
I don't think we need to be raising exceptions here.  You can add the validation for the number of samples in `check_request_factories` to ensure there's only one and add a error message there instead of throwing an exception and having to catch it again.  You can probably do away with the `@raised_exceptions` workflow then.

Or, even simpler, don't accept an array of samples.  We can make GraphQL only accept one sample.